src/proof/nova.rs

```diff
@@ -1472,72 +1472,70 @@ mod tests {
 
     #[test]
     #[ignore]
-    fn outer_prove_evaluate_zero_arg_lambda1() {
+    fn outer_prove_evaluate_zero_arg_lambda() {
         let s = &mut Store::<Fr>::default();
         let expected = s.num(123);
         let terminal = s.get_cont_terminal();
         nova_test_aux(
             s,
-            "((lambda () 123))",
+            "((lambda(x) x) 123)",
             Some(expected),
             None,
             Some(terminal),
             None,
-            3,
+            4,
         );
     }
 
     #[test]
     #[ignore]
     fn outer_prove_evaluate_zero_arg_lambda2() {
         let s = &mut Store::<Fr>::default();
-        let expected = s.num(10);
+        let expected = s.num(123);
         let terminal = s.get_cont_terminal();
         nova_test_aux(
             s,
-            "(let ((x 9) (f (lambda () (+ x 1)))) (f))",
+            "((lambda (y) ((lambda (x) y) 321)) 123)",
             Some(expected),
             None,
             Some(terminal),
             None,
-            10,
+            9,
         );
     }
+
     #[test]
-    fn outer_prove_evaluate_zero_arg_lambda3() {
+    fn outer_prove_evaluate_lambda3() {
         let s = &mut Store::<Fr>::default();
-        let expected = {
-            let arg = s.sym("x");
-            let num = s.num(123);
-            let body = s.list(&[num]);
-            let env = s.nil();
-            s.intern_fun(arg, body, env)
-        };
+        let expr = "((lambda (y) ((lambda (x) ((lambda (z) z) x)) y)) 123)";
+
+        let expected = s.num(123);
         let terminal = s.get_cont_terminal();
-        nova_test_aux(
-            s,
-            "((lambda (x) 123))",
-            Some(expected),
-            None,
-            Some(terminal),
-            None,
-            3,
-        );
+        nova_test_aux(s, expr, Some(expected), None, Some(terminal), None, 10);
```

Comment:
This clearly changes the purpose of the test. As above, if you want to add new tests, that's fine — but please don't remove important tests of correctness. The previous test was checking that the result of calling a one-arg function with zero args is the function itself. The one you've replaced it with has no relationship to that.